lib/std/fs.zig

```diff
@@ -1823,7 +1823,7 @@ pub const Dir = struct {
         var atomic_file = try dest_dir.atomicFile(dest_path, .{ .mode = mode });
         defer atomic_file.deinit();
 
-        try atomic_file.file.writeFileAll(in_file, .{ .in_len = size });
+        try os.copy_file(in_file.handle, atomic_file.file.handle, .{});
```

Comment:
The docs for `in_len` mention:

>         /// If the size of the source file is known, passing the size here will save one syscall.
>        in_len: ?u64 = null,

Did we gain a redundant fstat syscall in the case that sendfile has to be used? I know we've had this discussion before, and your position is that it doesn't matter in practice, but this is fundamental to zig's premise as a language and standard library, that it does the "optimal" interaction with the OS. I think it's an important sign of implementation quality for an strace / ProcMon session to contain only necessary syscalls.

Reply:
> Did we gain a redundant fstat syscall in the case that sendfile has to be used? 

No, both `copy_file_range` and `sendfile` are asked to read/write as many bytes as possible at each iteration. This saves a stat() (that's done anyway if when the file mode is copied over, I'd expect `copyFile` to copy the timestamps too but that's a different topic)